File: src/PlannedUnit.java
public class PlannedUnit {
    private Unit unit;
    private TeachingPeriod teachingPeriod;
    private boolean force; //if true, student can take this unit regardless of other requirements
    int year; //the year the student plans to take this unit

    public PlannedUnit(Unit unit, TeachingPeriod teachingPeriod, int year, boolean force) {
        this.unit = unit;
        this.teachingPeriod = teachingPeriod;
        this.year = year;
        this.force = force;
        
    }

    //getters
    public Unit getUnit() {
        return unit;
    }

    public TeachingPeriod getTeachingPeriod() {
        return teachingPeriod;
    }

    public boolean isForced() {
        return force;
    }

    public int getYear() {
        return year;
    }

    //setters
    public void setUnit(Unit unit) {
        this.unit = unit;
    }

    public void setTeachingPeriod(TeachingPeriod teachingPeriod) {
        this.teachingPeriod = teachingPeriod;
    }

    public void setForce(boolean force) {
        this.force = force;
    }

    public void setYear(int year) {
        this.year = year;
    }

}

File: src/planned_unit.cpp
#include "planned_unit.hpp"

#include "unit.hpp"

namespace course_planner {

PlannedUnit::PlannedUnit(const Unit& unit, TeachingPeriod teaching_period,
                         int year, bool force)
    : unit_(&unit),
      teaching_period_(teaching_period),
      year_(year),
      force_(force) {}

const Unit& PlannedUnit::unit() const {
    return *unit_;
}

TeachingPeriod PlannedUnit::teaching_period() const {
    return teaching_period_;
}

bool PlannedUnit::is_forced() const {
    return force_;
}

int PlannedUnit::year() const {
    return year_;
}

void PlannedUnit::set_unit(const Unit& unit) {
    unit_ = &unit;
}

void PlannedUnit::set_teaching_period(TeachingPeriod teaching_period) {
    teaching_period_ = teaching_period;
}

void PlannedUnit::set_force(bool force) {
    force_ = force;
}

void PlannedUnit::set_year(int year) {
    year_ = year;
}

}  // namespace course_planner
